### PR TITLE
feat(combatzone): report the groups that can no longer fight

### LINKED ISSUE #177: système pour déterminer quand un groupe est inopérationnel
Au niveau du spawn, avec une config, et par défaut l'utilisation de la desc de DCS (ex: "SAM TR" = true)
On détermine des ensembles d'unités importantes et quand elles sont pétées c'est tout le groupe qui est considéré comme inop.

    local groupsImportantUnits = {
        [".*s300.*"] = {
            minimumLife = 80,
            importantSets = {
                ["TR"] = { "S-300PS 40B6M tr" },
                ["SR"] = { "S-300PS 40B6MD sr", "S-300PS 64H6E sr" },
                ["CP"] = { "S-300PS 54K6 cp" }
            }
        }
    }

### PR DESCRIPTION
**Pull request by davidp57**
Closes [#177](https://github.com/VEAF/VEAF-Mission-Creation-Tools/issues/177), from 2023.
[`FEAT-GROUP-COMBAT-INEFFECTIVE`](.backlog/FEAT-GROUP-COMBAT-INEFFECTIVE/PRD.md).

A group is not only alive or dead. An S-300 whose tracking radar is destroyed keeps its launchers, its
trucks and its crew, counts as alive everywhere in this codebase, and in play is finished.

```
OUT OF ACTION (can no longer fight): ALPHA-SA10
```

## The predicate

`veaf.isGroupCombatEffective(groupOrName)`, two paths:

**A pattern table** (`veaf.ImportantUnitsByGroupPattern`) declares the sets of units a kind of site
cannot do without. Losing a whole set finishes the group. The S-300 entry from the issue body ships as
the first one:

```lua
[".*s300.*"] = {
  minimumLife = 80,
  importantSets = {
    TR = { "S-300PS 40B6M tr" },
    SR = { "S-300PS 40B6MD sr", "S-300PS 64H6E sr" },
    CP = { "S-300PS 54K6 cp" },
  },
}
```

`minimumLife` is a **percentage**, compared through the existing `veaf.getUnitLifeRelative`. Absolute hit
points would mean a different threshold per unit type, which no mission maker can reason about.

**With no pattern, the DCS attributes decide.** A group with a living `SAM SR` or `SAM LL` *is* a SAM
site, and is finished once nothing living carries `SAM TR`. A Tunguska is its own radar and launcher and
stays operational alone; a convoy has no radar and is a threat while it rolls.

Attributes are read from **our own generated `dcsUnits` database**, not `Unit.getDesc()`: same data, no
DCS call, and a unit type can be asked about without a living unit to ask through — which is what makes
this testable at all. One test checks the **real** database still carries the attributes the rule reads,
because a regeneration that dropped `SAM TR` would quietly declare every SAM site finished, mission-wide.

## The limit, stated rather than hidden

Destroyed units vanish from `Group:getUnits()`, so the predicate **cannot know a group had a radar it has
since lost**. The pattern table carries that knowledge — a pattern asserts the group owns those sets. The
default cannot, so a SAM site stripped of both its radars *and* its launchers reads as an ordinary group.
By then nothing dangerous is left of it, which is why that is acceptable rather than merely tolerated.

## One adopter, on purpose

The PRD warned that `completionCheck` is *"the visible one and therefore the riskiest"* — adopting it
changes when zones complete, in every existing mission. So the first adopter is the **F10 report**, which
adds information and removes none. **No mission behaviour changes.**

A wiped-out group is reported as destroyed by the existing tallies, not as out of action. The predicate
answers false for both, so the report tells them apart by checking for survivors itself — a test pins
that, because naming a dead group would put noise on every report for the rest of the mission.

### What adopting it elsewhere would change, for a player

| | Today | With the predicate in `completionCheck` |
|---|---|---|
| Kill an S-300's tracking radar and leave | zone stays open; hunt every launcher and truck | zone completes |
| A zone with a SAM site **and** a convoy | completes when both are gone | completes when the convoy is gone and the SAM is toothless |
| A toothless but intact SAM | stays open | **completes with units still alive and visible** |

That last row is a **design question, not a technical one**: a zone announcing "all enemies destroyed"
while a player sees four intact launchers is either a welcome shortcut or a bug, depending on intent. It
also changes every existing mission holding a SAM site. Filed as
[`FEAT-COMBAT-EFFECTIVE-ADOPTION`](.backlog/FEAT-COMBAT-EFFECTIVE-ADOPTION/PRD.md) with the arbitration
left to David, and Skynet named as the safer next step (`findSkynetElementToDefend` skipping dead sites
cannot end a mission early).

## Two things found while building it

**The predicate first re-filtered `getUnits()` through `veaf.isUnitAlive`** — redundant, since DCS
already drops a destroyed unit from its group, and it imposed an `isExist`/`isActive` requirement that
broke four existing tests whose unit stubs quite reasonably lacked them. Removed rather than worked
around by patching the stubs.

**Lua coverage jumped from 75% to 83%, and it was false.** The reality test loads `dcsUnits.lua` — 13 600
lines of generated literals, all counted as covered. Left alone, that inflation would have been baked
into the ratchet floor and collapsed the moment the load went away. The file is now excluded from
coverage, and the real figure is **75.18%**.

## Tests

`test_veaf` 398 → **417**: the pattern path (complete site, each set lost in turn, life below and exactly
at the threshold, one survivor per set, case-insensitive matching), the attribute default (site without a
tracker, site keeping one, self-contained SAM, convoy), degenerate inputs (empty group, unknown name, nil,
group-by-name, unknown unit type), and three checks against the real database.

`test_veafCombatZone` 234 → **239**: an effective zone reads as before, an ineffective group is named,
several are all named, a destroyed group is *not* named, and an inactive zone still says nothing.

## Checks

| Gate | Result |
|---|---|
| `poetry run test-lua` | **ALL 37 SUITE(S) PASSED** |
| Lua coverage | **75.18%** against a 74 floor — 1.18 pt gap, no bump due |
| `poetry run pytest` | **3815 passed**, 5 skipped, exit 0, on the settled tree |
| `stylua --check` | clean |
| `poetry run docs-check` | no defect found |

`luacheck` is not installed on this machine; the CI Lua gate covers it. Diff 32 947 characters.

Documented in both languages on the `veafCombatZone` page, including the explicit note that this does
**not** change when a zone ends.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Distinguish surviving groups that are no longer combat-effective and expose them in zone reports without changing mission completion behavior.

New Features:
- Add combat-effectiveness detection for groups using configurable group patterns and DCS unit attributes.
- Report surviving groups that can no longer fight in combat-zone F10 information messages.

Bug Fixes:
- Exclude the generated DCS unit database from Lua coverage so coverage metrics reflect executable logic accurately.

Enhancements:
- Document the combat-effectiveness rules, their limitations, and planned adoption options for other systems.

Build:
- Bump the project version to 6.15.24.

Documentation:
- Document out-of-action group reporting and its unchanged zone-completion behavior in English and French.

Tests:
- Add Lua coverage for pattern-based and attribute-based combat effectiveness, database integrity, edge cases, and combat-zone reporting.

Chores:
- Record the completed feature and its follow-up adoption work in the backlog.

### CODE REVIEW

File: .backlog/FEAT-COMBAT-EFFECTIVE-ADOPTION/PRD.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# FEAT-COMBAT-EFFECTIVE-ADOPTION — adopt the combat-effectiveness predicate beyond the report
+
+Status: ⬜ ready
+
+Follow-up to [`FEAT-GROUP-COMBAT-INEFFECTIVE`](../FEAT-GROUP-COMBAT-INEFFECTIVE/PRD.md), which shipped
+`veaf.isGroupCombatEffective` in 6.15.24 and deliberately adopted it in **one** place — the F10 zone
+report, because that changes no mission behaviour. The analysis of the other two callers lives in that
+PRD; this lot is the acting on it.
```
[the remark this answers is not in the document]
Comment (davidp57) [reply .backlog/FEAT-COMBAT-EFFECTIVE-ADOPTION/PRD.md]: Fixed in 7c40bda6 — "this lot is the acting on it" was simply wrong. Went with "this lot acts on it", shorter than either suggested form and the same meaning.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
